cpp/src/jni/dataset/jni_util.cc

```diff
@@ -162,31 +194,40 @@ std::shared_ptr<ReservationListener> ReservationListenableMemoryPool::get_listen
 
 ReservationListenableMemoryPool::~ReservationListenableMemoryPool() {}
 
+Status CheckException(JNIEnv* env) {
+  if (env->ExceptionCheck()) {
+    env->ExceptionDescribe();
+    env->ExceptionClear();
+    return Status::Invalid("Error during calling Java code from native code");
+  }
+  return Status::OK();
+}
+
 jclass CreateGlobalClassReference(JNIEnv* env, const char* class_name) {
   jclass local_class = env->FindClass(class_name);
   jclass global_class = (jclass)env->NewGlobalRef(local_class);
   env->DeleteLocalRef(local_class);
   return global_class;
 }
 
-arrow::Result<jmethodID> GetMethodID(JNIEnv* env, jclass this_class, const char* name,
```

Comment:
We would like to ask you to avoid these types of the cleanup in this huge PR.  Increasing # of changed lines makes us harder to review this PR.

What do you think?

Reply:
Agreed. The Nit changes was now removed from this PR. I'll make another :)